enhancements/installer/aws-custom-dns.md

```diff
@@ -0,0 +1,475 @@
+}
+
+type Platform struct {
+// <snip>
+
+// All configuration required for enabling the optional customDNS
+CustomDNS CustomDNS `json:"customDNS,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+### Risks and Mitigations
+
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions [optional]
+
+One of the decisions that needs to be made early in the design process is to
+determine the best way to order OpenShift's configuration of the Load Balancer
+and the customer's configuration of their custom DNS solution.
+
+Today, the Installer configures the API's LB first and the generated DNS name
+is used to configure DNS CNAME records. The LB DNS name cannot be predicted in
+advance because it is generated using an unique UUID while configuring LB
+entries for API, API-Int and Ingress services.
+
+Here are a few options being considered as potential solutions:
+
+#### Option 1: Custom DNS configuration before creating the cluster
+
+In this Option, the admin manually configures the Custom DNS solution before
+cluster creation is started. The Custom DNS is configured with an A record
+that points to static IP addresses within the public subnets of the machine
+network.
+
+When users provide their own subnets during install, they are responsible
+for creating the DNS records for the API servers using the IP addresses for
+NICs on the pre-created public subnets. Similarly, the IP addresses for NICs
+on the private subnets can be provided to the MCO via the Infrastruture CR.
+MCO uses these IP addresses to configure DNS for the Internal API Service.
+
+Is there a scenario when the public and private subnets are not created in
+advance? In that case, should the user be providing the API and API-Int IP
+addresses via install-config?
+
+This [RFE](https://access.redhat.com/solutions/4846611) talks about only using
+CNAME records for DNS configuration. How should this option be modified to
+handle this RFE?
+
+#### Option 2: Custom DNS configuration after creating the cluster
+
+In this Option, the admin creates the OpenShift cluster first. The CoreDNS pods
+provide DNS resolution for both the Internal API and Ingress services. After
+the cluster is created successfully, the Installer generates as its output the
+LB DNS name for the API Server that the admin can use to configure the custom
+DNS with a CNAME record that points to this LB DNS name.
+
+The Installer could also output the LB DNS name for the Ingress service and the
+custom DNS is optionally (the CoreDNS pods already provide DNS resolution for
+Ingress) configured with this too.
+
+#### Option 3: BYO DNS and LB
+
+This solution requires the customer to configure Load Balancers on their own and
+use the generated DNS names to configure their DNS solution before they run
+"openshift-install create cluster"
+
+This solution will have additional changes/implications in other components of
+OpenShift that this enhancement hasn't tackled.
```

Comment:
To me this seems like an extension of Option 1. So in addition to users providing their own subnets, they would also configure their own LBs on those subnets and the installer would skip creating subnets.

The benefit to this approach is that the LB DNS name would be known pre-install. But this would require additional effort (essentially implementing a BYO LB feature in addition to BYO DNS).

Reply:
In the case of Installer created subnets, we will not be able to pre configure the LB and DNS. Upon further discussion and investigation, one clear solution has emerged. So, removing these options in the latest update. PTAL.